ios_config.sh

```diff
@@ -70,45 +72,47 @@ echo "info: 1) Locating ${_JSON_FILE_NAME} file:"
 
 if [[ -z ${_CURRENT_SEARCH_DIR} ]]; then
   _CURRENT_SEARCH_DIR=$(pwd)
-fi;
+fi
 
 while true; do
   _CURRENT_SEARCH_DIR=$(dirname "$_CURRENT_SEARCH_DIR")
 
   if [[ "$_CURRENT_SEARCH_DIR" == "/" ]] || [[ ${_CURRENT_LOOKUPS} -gt ${_MAX_LOOKUPS} ]]; then
-    break;
-  fi;
+    break
+  fi
 
-  echo "info:      ($_CURRENT_LOOKUPS of $_MAX_LOOKUPS) Searching in '$_CURRENT_SEARCH_DIR' for a ${_JSON_FILE_NAME}/${_JS_APP_CONFIG_FILE_NAME} file."
+  echo "info:      ($_CURRENT_LOOKUPS of $_MAX_LOOKUPS) Searching in '$_CURRENT_SEARCH_DIR' for a ${_JSON_FILE_NAME}/${_JS_APP_CONFIG_FILE_NAME}/${_TS_APP_CONFIG_FILE_NAME} file."
 
-  _SEARCH_RESULT=$(find "$_CURRENT_SEARCH_DIR" -maxdepth 2 \( -name ${_JSON_FILE_NAME} -o -name ${_JS_APP_CONFIG_FILE_NAME} \) -print | /usr/bin/head -n 1)
+  _SEARCH_RESULT=$(find "$_CURRENT_SEARCH_DIR" -maxdepth 2 \( -name ${_JSON_FILE_NAME} -o -name ${_JS_APP_CONFIG_FILE_NAME} -o -name ${_TS_APP_CONFIG_FILE_NAME} \) -print | /usr/bin/head -n 1)
 
-  if [[ "$(basename ${_SEARCH_RESULT})" = "${_JS_APP_CONFIG_FILE_NAME}" ]]; then
+  if [[ "$(basename ${_SEARCH_RESULT})" = "${_JS_APP_CONFIG_FILE_NAME}" ]] || [[ "$(basename ${_SEARCH_RESULT})" = "${_TS_APP_CONFIG_FILE_NAME}" ]]; then
     _IS_CONFIG_JS=true
-    echo "info:      ${_JS_APP_CONFIG_FILE_NAME} found at $_SEARCH_RESULT"
-    break;
-  fi;
+    echo "info:     expo config found at $_SEARCH_RESULT"
+    break
+  fi
 
   if [[ "$(basename ${_SEARCH_RESULT})" = "${_JSON_FILE_NAME}" ]]; then
     echo "info:      ${_JSON_FILE_NAME} found at ${_SEARCH_RESULT}"
-    break;
-  fi;
+    break
+  fi
 
-  _CURRENT_LOOKUPS=$((_CURRENT_LOOKUPS+1))
+  _CURRENT_LOOKUPS=$((_CURRENT_LOOKUPS + 1))
 done
 
 if [[ ${_SEARCH_RESULT} ]]; then
   if [[ ${_IS_CONFIG_JS} == "true" ]]; then
-    _JSON_OUTPUT_RAW=$("${NODE_BINARY}" -e "console.log(JSON.stringify(require('${_SEARCH_RESULT}')));")
+    _APP_CONFIG_DIR_NAME=$(dirname $_SEARCH_RESULT)
+    _JSON_OUTPUT_RAW=$("${NODE_BINARY}" -e "try { const expoConfig = require('@expo/config'); console.log(JSON.stringify(expoConfig.getConfig('${_APP_CONFIG_DIR_NAME}')?.exp?.extra));  } catch (e) { if (e instanceof Error && e.code === 'MODULE_NOT_FOUND') { console.error('You have an expo app.config[.js|.ts] file but do not have @expo/config installed. @expo/config is required for react-native-google-mobile-ads to read the app.config[.js|.ts] file and should be installed by default with expo. Please install @expo/config and try again.') } else { throw e }}")
```

Comment:
```suggestion
    _JSON_OUTPUT_RAW=$("${NODE_BINARY}" -e "try { const expoConfig = require('@expo/config'); console.log(JSON.stringify(expoConfig.getConfig('${_APP_CONFIG_DIR_NAME}')?.expo?.extra));  } catch (e) { if (e instanceof Error && e.code === 'MODULE_NOT_FOUND') { console.error('You have an expo app.config[.js|.ts] file but do not have @expo/config installed. @expo/config is required for react-native-google-mobile-ads to read the app.config[.js|.ts] file and should be installed by default with expo. Please install @expo/config and try again.') } else { throw e }}")
```

typo?

Reply:
Not a typo. The @expo/config package for whatever reason exposes the config under the `exp` key.